upup/pkg/fi/lifecycle.go

```diff
@@ -39,3 +41,12 @@ const (
 type HasLifecycle interface {
 	GetLifecycle() *Lifecycle
 }
+
+// Phases are used for validation and cli help.
+var Lifecycles = sets.NewString(
```

Comment:
Maybe use `[]fi.Lifecycle`, so we don't give up type safety?  We might need another helper or two, but that's OK.

Reply:
Unfortunately, we cannot do that since I am getting a cycle in dependency graph.

```
ERROR: /Users/clove/Workspace/src/k8s.io/kops/upup/pkg/fi/BUILD.bazel:3:1: in go_library rule //upup/pkg/fi:go_default_library: cycle in dependency graph:
    //cmd/nodeup:go_default_library
    //nodeup/pkg/bootstrap:go_default_library
    //upup/pkg/fi/nodeup/nodetasks:go_default_library
.-> //upup/pkg/fi:go_default_library [self-edge]
`--
```

Reply:
I think I am wrong ... checking

Reply:
The only place we are using this is to output the lifecycle names in the CLI UX.  Because of that, we are not needing type safety.  Is there another place that you assumed was using this?